okhttp/src/main/java/okhttp3/internal/http2/Http2Stream.java

```diff
@@ -153,7 +156,8 @@ public synchronized Headers takeHeaders() throws IOException {
     if (!headersQueue.isEmpty()) {
       return headersQueue.removeFirst();
     }
-    throw new StreamResetException(errorCode);
+    if (errorException == null) throw new StreamResetException(errorCode);
+    throw errorException;
```

Comment:
nit: consider making this and the other occurrences more concise:
```
throw errorException != null ? errorException : new StreamResetException(errorCode);
```

Reply:
Fixed.